src/update_engine/delta_performer.cc

```diff
@@ -461,10 +464,18 @@ bool DeltaPerformer::PerformReplaceOperation(
   // point to one of the ExtentWriter objects above.
   ExtentWriter* writer = NULL;
   if (operation.type() == DeltaArchiveManifest_InstallOperation_Type_REPLACE) {
-    writer = &zero_pad_writer;
+    if (is_kernel_partition) {
```

Comment:
Perhaps should be changed to `is_kernel` but we can do cosmetic stuff like that later too, in a PR that doesn't include functionality changes.